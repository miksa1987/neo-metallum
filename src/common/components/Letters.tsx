import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { alphabets } from '../constants';

interface Props {
  path: string
}

const Layout = styled.div`
  display: flex;
  flex-wrap: wrap;

  background-color: #44475a;
  border: 1px solid black;

  @media screen and (max-width: 1050px) {
    min-height: 7rem;
  }

  @media screen and (max-width: 750px) {
    min-height: 10rem;
  }

  @media screen and (max-width: 650px) {
    min-height: 13rem;
  }

  @media screen and (max-width: 750px) {
    min-height: 7rem;
  }
`;

const StyledLink = styled(Link)`
  margin-right: 1rem;
  font-size: 150%;
`;

export const Letters: React.FC<Props> = ({ path }) => {
  return (
    <Layout>
      {alphabets.map((letter, i) =>
        <StyledLink
          id={`${letter}-link`}
          key={i} 
          to={`${path}/${letter}`}
        >
          {letter.toUpperCase()}
        </StyledLink>)}
    </Layout>
  );
}