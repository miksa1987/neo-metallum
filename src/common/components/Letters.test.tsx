import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';

import { Letters } from './Letters';

describe('Letters', () => {
  afterEach(cleanup);

  test('Snapshot passes', () => {
    const history = createMemoryHistory();
    const component = render(
      <Router history={history}>
        <Letters path='/test' />
      </Router>
    );
    expect(component).toMatchSnapshot();
  });

  test('URL is manipulated correctly', () => {
    const history = createMemoryHistory();
    const component = render(
      <Router history={history}>
        <Letters path='/test' />
      </Router>
    );

    const aLink: any = component.container.querySelector('#a-link');
    fireEvent.click(aLink);

    setTimeout(() => {
      expect(history.location.pathname).toBe('/test/a');
    }, 100);
  });
});