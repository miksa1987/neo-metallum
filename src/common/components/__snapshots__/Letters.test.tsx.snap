// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Letters Snapshot passes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxjAm kDxRMj"
      >
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/a"
          id="a-link"
        >
          A
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/b"
          id="b-link"
        >
          B
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/c"
          id="c-link"
        >
          C
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/d"
          id="d-link"
        >
          D
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/e"
          id="e-link"
        >
          E
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/f"
          id="f-link"
        >
          F
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/g"
          id="g-link"
        >
          G
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/h"
          id="h-link"
        >
          H
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/i"
          id="i-link"
        >
          I
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/j"
          id="j-link"
        >
          J
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/k"
          id="k-link"
        >
          K
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/l"
          id="l-link"
        >
          L
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/m"
          id="m-link"
        >
          M
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/n"
          id="n-link"
        >
          N
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/o"
          id="o-link"
        >
          O
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/p"
          id="p-link"
        >
          P
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/q"
          id="q-link"
        >
          Q
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/r"
          id="r-link"
        >
          R
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/s"
          id="s-link"
        >
          S
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/t"
          id="t-link"
        >
          T
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/u"
          id="u-link"
        >
          U
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/v"
          id="v-link"
        >
          V
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/w"
          id="w-link"
        >
          W
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/x"
          id="x-link"
        >
          X
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/z"
          id="z-link"
        >
          Z
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/å"
          id="å-link"
        >
          Å
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/ä"
          id="ä-link"
        >
          Ä
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/ö"
          id="ö-link"
        >
          Ö
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/0"
          id="0-link"
        >
          0
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/1"
          id="1-link"
        >
          1
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/2"
          id="2-link"
        >
          2
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/3"
          id="3-link"
        >
          3
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/4"
          id="4-link"
        >
          4
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/5"
          id="5-link"
        >
          5
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/6"
          id="6-link"
        >
          6
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/7"
          id="7-link"
        >
          7
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/8"
          id="8-link"
        >
          8
        </a>
        <a
          class="sc-AxirZ cVaVsq"
          href="/test/9"
          id="9-link"
        >
          9
        </a>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm kDxRMj"
    >
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/a"
        id="a-link"
      >
        A
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/b"
        id="b-link"
      >
        B
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/c"
        id="c-link"
      >
        C
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/d"
        id="d-link"
      >
        D
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/e"
        id="e-link"
      >
        E
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/f"
        id="f-link"
      >
        F
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/g"
        id="g-link"
      >
        G
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/h"
        id="h-link"
      >
        H
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/i"
        id="i-link"
      >
        I
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/j"
        id="j-link"
      >
        J
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/k"
        id="k-link"
      >
        K
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/l"
        id="l-link"
      >
        L
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/m"
        id="m-link"
      >
        M
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/n"
        id="n-link"
      >
        N
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/o"
        id="o-link"
      >
        O
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/p"
        id="p-link"
      >
        P
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/q"
        id="q-link"
      >
        Q
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/r"
        id="r-link"
      >
        R
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/s"
        id="s-link"
      >
        S
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/t"
        id="t-link"
      >
        T
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/u"
        id="u-link"
      >
        U
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/v"
        id="v-link"
      >
        V
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/w"
        id="w-link"
      >
        W
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/x"
        id="x-link"
      >
        X
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/z"
        id="z-link"
      >
        Z
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/å"
        id="å-link"
      >
        Å
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/ä"
        id="ä-link"
      >
        Ä
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/ö"
        id="ö-link"
      >
        Ö
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/0"
        id="0-link"
      >
        0
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/1"
        id="1-link"
      >
        1
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/2"
        id="2-link"
      >
        2
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/3"
        id="3-link"
      >
        3
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/4"
        id="4-link"
      >
        4
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/5"
        id="5-link"
      >
        5
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/6"
        id="6-link"
      >
        6
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/7"
        id="7-link"
      >
        7
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/8"
        id="8-link"
      >
        8
      </a>
      <a
        class="sc-AxirZ cVaVsq"
        href="/test/9"
        id="9-link"
      >
        9
      </a>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
